resources/user.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Cookbook Name:: sensu-go
+# Resource:: agent
+#
+# Copyright:: 2018 Sensu, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+resource_name :sensu_user
+
+action_class do
+  include SensuCookbook::Helpers
+end
+
+property :config_home, String, default: '/etc/sensu'
+
+property :password, String
+property :roles, Array
+property :disabled, [true, false], default: false
+
+action :create do
+  directory object_dir do
+    action :create
+    recursive true
+  end
+
+  execute "execute[sensuctl user create #{new_resource.name} -p #{new_resource.password} -r #{new_resource.roles}]" do
```

Comment:
Hmm, the problem with this is that this exposes the password to the process table which is readable by any user on the system.

Reply:
I think execute with $PASSWORD and env setting the password would address this. Is that correct?

Reply:
Yeah, I didn't think this was the best way to do this, so I'd love some alternate suggestions.

Reply:
if you want a valid solution that requires little change you can write it out to a file and then do:
```
execute "execute[sensuctl user create #{new_resource.name} -p $(cat PATH_TO_CREDS_FILE) -r #{new_resource.roles}]" do
```